File: ios/Source/Models/EventType.swift
// Copyright (c) 2018-2019 Coinbase, Inc. <https://coinbase.com/>
// Licensed under the Apache License, version 2.0

import Foundation

enum EventType: String, Codable {
    /// Web3 related requests
    case web3Request = "Web3Request"

    /// Web3 related responses
    case web3Response = "Web3Response"

    /// Web3 request has been canceled by the host
    case web3RequestCanceled = "Web3RequestCanceled"
}

File: android/walletlink/src/main/java/com/coinbase/walletlink/models/EventType.kt
// Copyright (c) 2018-2019 Coinbase, Inc. <https://coinbase.com/>
// Licensed under the Apache License, version 2.0

package com.coinbase.walletlink.models

import com.squareup.moshi.Json

internal enum class EventType(val rawValue: String) {
    /**
     * Web3 related requests
     */
    @Json(name = "Web3Request") Web3Request("Web3Request"),

    /**
     * Web3 related responses
     */
    @Json(name = "Web3Response") Web3Response("Web3Response"),

    /**
     * Web3 request has been canceled by the host
     */
    @Json(name = "Web3RequestCanceled") Web3RequestCanceled("Web3RequestCanceled");
}
